src/configure/helper/repoAnalysisHelper.ts

```diff
@@ -46,32 +47,35 @@ export class RepoAnalysisHelper {
                 let applicationSettings: RepositoryAnalysisApplicationSettings = new RepositoryAnalysisApplicationSettings();
                 applicationSettings.language = analysis.language;
 
-                if(!!analysis.settings && !!analysis.buildTargetName) {
-                    applicationSettings.buildTargetName = analysis.buildTargetName;
-                    if (analysis.language === SupportedLanguage.NODE) {
-                        applicationSettings.settings.nodePackageFilePath = analysis.settings[RepoAnalysisConstants.PackageFilePath];
-                        applicationSettings.settings.nodePackageFileDirectory = path.dirname(analysis.settings[RepoAnalysisConstants.PackageFilePath]);
-                        if (analysis.buildTargetName === RepoAnalysisConstants.Gulp && !!analysis.settings[RepoAnalysisConstants.GulpFilePath]) {
-                            applicationSettings.settings.nodeGulpFilePath = analysis.settings[RepoAnalysisConstants.GulpFilePath];
+                if(!!analysis.settings){
+                    applicationSettings.settings.workingDirectory = !!analysis.settings.workingDirectory ? analysis.settings.workingDirectory : workspacePath;
+                    if(!!analysis.buildTargetName) {
+                        applicationSettings.buildTargetName = analysis.buildTargetName;
+                        if (analysis.language === SupportedLanguage.NODE) {
+                            applicationSettings.settings.nodePackageFilePath = analysis.settings[RepoAnalysisConstants.PackageFilePath];
+                            applicationSettings.settings.nodePackageFileDirectory = path.dirname(analysis.settings[RepoAnalysisConstants.PackageFilePath]);
+                            if (analysis.buildTargetName === RepoAnalysisConstants.Gulp && !!analysis.settings[RepoAnalysisConstants.GulpFilePath]) {
+                                applicationSettings.settings.nodeGulpFilePath = analysis.settings[RepoAnalysisConstants.GulpFilePath];
```

Comment:
where are these being used? Shouldn't they be used while creating workflow? I have left a comment there as well.

Reply:
GitHub Templates